index.js

```diff
@@ -56,12 +56,12 @@ function interpolate(t, degree, points, knots, weights, result) {
     v[i][d] = weights[i];
   }
 
-  // l (level) goes from 1 to the curve degree
+  // l (level) goes from 1 to the curve degree + 1
   var alpha;
-  for(l=1; l<=degree; l++) {
+  for(l=1; l<=degree+1; l++) {
     // build level l of the pyramid
-    for(i=s; i>s-degree+l; i--) {
-      alpha = (t - knots[i]) / (knots[i+degree-l] - knots[i]);
+    for(i=s; i>s-degree-1+l; i--) {
```

Comment:
There's still something to be done to make the whole process simpler to understand, I think. Making `l` start at 0 would be a good start. I'd like to make the relationship between degree, knot count and control points more self-evident.

I'm thinking about making an un-optimized version of the algorithm that wouldn't be exported, to serve as a reference. I feel the inlining of the `i` loop and the reuse of `v[i]` create a lot of confusion.